packages/flp-config-inquirer/src/prompts/prompts.ts

```diff
@@ -58,27 +54,37 @@ export function getQuestions(
             silentOverwrite,
             promptOptions?.[promptNames.subTitle]
         ),
-        [promptNames.additionalParameters]: getParameterStringPrompt(
-            inboundKeys,
-            promptOptions?.[promptNames.additionalParameters]
-        ),
-        [promptNames.createAnotherInbound]: getCreateAnotherInboundPrompt(
-            isCLI,
-            promptOptions?.[promptNames.createAnotherInbound]
-        )
+        [promptNames.additionalParameters]: getParameterStringPrompt(inboundKeys),
+        [promptNames.createAnotherInbound]: getCreateAnotherInboundPrompt(isCLI)
     };
 
-    const questions: FLPConfigQuestion[] = [
-        keyedPrompts[promptNames.inboundId],
-        keyedPrompts[promptNames.emptyInboundsInfo],
-        keyedPrompts[promptNames.semanticObject],
-        keyedPrompts[promptNames.action],
-        keyedPrompts[promptNames.overwrite],
-        keyedPrompts[promptNames.title],
-        keyedPrompts[promptNames.subTitle],
-        keyedPrompts[promptNames.additionalParameters],
-        keyedPrompts[promptNames.createAnotherInbound]
-    ];
+    const questions: FLPConfigQuestion[] = [];
+
+    if (promptOptions?.[promptNames.inboundId]?.hide !== true) {
+        questions.push(keyedPrompts[promptNames.inboundId]);
+    }
+
+    if (promptOptions?.[promptNames.emptyInboundsInfo]?.hide !== true) {
+        questions.push(keyedPrompts[promptNames.emptyInboundsInfo]);
+    }
+
+    questions.push(
+        ...[
+            keyedPrompts[promptNames.semanticObject],
+            keyedPrompts[promptNames.action],
+            keyedPrompts[promptNames.overwrite],
+            keyedPrompts[promptNames.title],
+            keyedPrompts[promptNames.subTitle]
+        ]
+    );
+
+    if (promptOptions?.[promptNames.additionalParameters]?.hide !== true) {
+        questions.push(keyedPrompts[promptNames.additionalParameters]);
+    }
+
+    if (promptOptions?.[promptNames.createAnotherInbound]?.hide !== true) {
+        questions.push(keyedPrompts[promptNames.createAnotherInbound]);
+    }
```

Comment:
(optional) Alternatively, would it be better to map them and not repeat the same if statements?

```suggestion
    const questions: FLPConfigQuestion[] = Object.entries(keyedPrompts)
        .filter(([promptName]) => {
            const option = promptOptions?.[promptName as promptNames];
            return !hasHideOption(option) || !option.hide;
        })
        .map(([_, prompt]) => prompt);
```

And then the helper method:

```ts
/**
 * Type guard to check if the provided object has a `hide` property.
 *
 * @param option - The object to check.
 * @returns True if the object has a `hide` property of type `boolean`, otherwise false.
 */
function hasHideOption(option: any): option is { hide: boolean } {
    return option && typeof option.hide === 'boolean';
}
```

Here options are `any` type. If that is not okay that would mean you need to write all interfaces of the promptNames options, or create a union one.

Reply:
I've taken a slightly different approach but with the same result i.e not having to write if statements for each one.